### PR TITLE
Assignment 3 Code - Review

### PR DESCRIPTION
**Pull request by andreeaiulia13**
This is a website for a Food Ordering System.

The main changes are:

- migrate Assignment 1 to Rails
- use scss for stylesheets
- use haml for html files
- store the data in database
- use Active Storage
- deploy the site on Heroku

The site can be seen on this Heroku link: https://assignment3-food-ordering-sys.herokuapp.com/

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -1,29 +1,58 @@
 class ProductsController < ApplicationController
-  before_action :set_product, only:[:show, :edit, :update, :destroy]
+ before_action :set_product, only:[:show, :edit, :update, :destroy]
```
Comment (vaneaclaudia) [inline app/controllers/products_controller.rb]: please leave an empty space between `only:` and the array of values

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -1,29 +1,58 @@
...
   end
 
   def create
+    begin
+      product = Product.create!(product_params.except(:image))
+      product.image.attach(product_params[:image])
+      
+      redirect_to dashboard_path
+    
+    rescue => exception
```
Comment (vaneaclaudia) [inline app/controllers/products_controller.rb]: It seems that you are not using `exception` here. Also, you will capture StandardError and all of its subtypes by not specifying an exact type of error to rescue. I think here you should capture the error thrown by Active Record (RecordInvalid)

File: app/controllers/products_controller.rb [hunk 3]
```diff
@@ -1,29 +1,58 @@
...
+    @products = Product.all
   end
-  
+
   def new
+    @product = Product.new
   end
 
   def create
+    begin
```
Comment (vaneaclaudia) [inline app/controllers/products_controller.rb]: you don't need to use `begin` keyword when you are doing the rescue for the whole implementation

**Review by vaneaclaudia — CHANGES_REQUESTED** [review]

File: app/controllers/products_controller.rb [hunk 4]
```diff
@@ -1,29 +1,57 @@
...
 
   def create
+    product = Product.create!(product_params.except(:image))
+    product.image.attach(product_params[:image])
+      
+    redirect_to dashboard_path
+    
+    rescue ActiveRecord::RecordInvalid
+      redirect_to new_error_product_path
+
```
Comment (vaneaclaudia) [inline app/controllers/products_controller.rb]: remove the empty line here. Also indentation looks a bit weird. `rescue` should be aligned with `def`

**Review by vaneaclaudia — APPROVED** [review]

File: app/assets/stylesheets/dashboard.scss [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  padding-bottom: 5px;
+  font-size: 23px;
+}
+
+.show-image {
+  margin-top: 25px;
+  border: rgb(131, 129, 129) solid 0.6px;
+  border-radius: 10%;
+  object-fit: cover;
+}
```
Comment (chioreandan) [inline app/assets/stylesheets/dashboard.scss:92]: Please add an empty line here, like you did on all the classes. :D

File: app/assets/stylesheets/products.scss [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+}
+
+body {
+  padding: 0;
+  overflow-x: hidden;
+  width: 100%;
+  height: 100%;
+}
+
+.add_link {
```
Comment (chioreandan) [inline app/assets/stylesheets/products.scss]: ```add-link```

**Review by chioreandan — APPROVED** [review]
Just 2 small changes and it is good to go 👍

**Pull request merged**